lib/ts/recipe/dashboard/api/userroles/roles/getAllRoles.ts

```diff
@@ -1,20 +1,32 @@
 import UserRoles from "../../../../userroles";
+import STError from "../../../../../error";
 import UserRolesRecipe from "../../../../userroles/recipe";
 
 import { APIFunction, APIInterface, APIOptions } from "../../../types";
 
-type Roles = Array<{ role: string; permissions: string[] }>;
+type Roles = { role: string; permissions: string[] }[];
 
 type Response =
     | {
           status: "OK";
+          totalPages: number;
+          totalRolesCount: number;
```

Comment:
this API should just return all the list of roles.. the pagination stuff happens on the frontend only.